src/main/java/io/cryostat/net/AuthManager.java

```diff
@@ -37,18 +37,25 @@
  */
 package io.cryostat.net;
 
+import java.util.Optional;
 import java.util.Set;
+import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Future;
 import java.util.function.Function;
 import java.util.function.Supplier;
 
 import io.cryostat.net.security.ResourceAction;
+import io.cryostat.net.web.http.api.v2.IntermediateResponse;
 
 public interface AuthManager {
     AuthenticationScheme getScheme();
 
     Future<UserInfo> getUserInfo(Supplier<String> httpHeaderProvider);
 
+    Optional<IntermediateResponse<UserInfo>> sendLoginRedirectIfRequired(
```

Comment:
Hmm... do we need to include the `IntermediateResponse` type in this interface? I would prefer to keep that used solely within HTTP handlers (within the `.http.api` subpackage) if possible.

The `UserInfo` generic type parameter looks like it isn't really needed here since the actual `IntermediateResponse` returned by the concrete implementation doesn't actually set a `UserInfo` response body object, and I think the `302` status code can simply be inferred by the Handler implementation if this method returns a non-empty `Optional`. So, the only required information actually encoded here is some HTTP headers, which can be represented with either just the `String` for the redirect URL (and the `X-Location` inferred by the Handler), or a simple struct-like object if more than one Header value needs to be returned.